dev/MRTCore/mrt/Core/src/MRM.cpp

```diff
@@ -28,12 +28,55 @@ typedef struct _MrmObjects
 } MrmObjects;
 
 constexpr wchar_t ResourceUriPrefix[] = L"ms-resource://";
-constexpr int ResourceUriPrefixLength = ARRAYSIZE(ResourceUriPrefix) - 1;
+constexpr unsigned int ResourceUriPrefixLength = ARRAYSIZE(ResourceUriPrefix) - 1;
 constexpr wchar_t c_defaultPriFilename[] = L"resources.pri";
 
 #define INDEX_RESOURCE_ID -1
 #define INDEX_RESOURCE_URI -2
 
+bool IsResourceUri(_In_ PCWSTR resourceId)
+{
+    return ((wcslen(resourceId) > static_cast<size_t>(ResourceUriPrefixLength)) && (CompareStringOrdinal(ResourceUriPrefix, ResourceUriPrefixLength, resourceId, ResourceUriPrefixLength, TRUE) == CSTR_EQUAL));
+}
+
+HRESULT GetRootAndRelativeIdFromUri(
+    _In_ PCWSTR resourceUri,
+    _Out_writes_z_(rootSize) PWSTR root,
+    unsigned int rootSize,
+    _Out_ unsigned int* indexOfRelatvieId)
+{
+    // We should have checked resourceUri length is longer than ResourceUriPrefixLength before calling this
```

Comment:
Wondering where we do this check?

Reply:
The function is always called after IsResourceUri. The check is done in IsResourceUri.